packages/babel-generator/src/generators/statements.js

```diff
@@ -123,10 +123,14 @@ function buildLabelStatement(prefix, key = "label") {
     const label = node[key];
     if (label) {
       this.space();
-
-      const terminatorState = this.startTerminatorless();
-      this.print(label, node);
-      this.endTerminatorless(terminatorState);
+      if (prefix == "break" || prefix == "continue") {
+        // see https://git.io/vQ7uG
+        this.print(label, node);
```

Comment:
My concern is that where before you'd get a very obvious syntax error when loading the code, now `babel-generator` will output code that will parse, but have different runtime behavior. That _probably_ will trigger a syntax error too, since there probably isn't a variable with the same name as the label, but it isn't guaranteed.

I don't care if we get a 100% fix in for this, but I do wonder if there is a way we could make `babel-generator` throw a clear error when that case happens, instead of outputting broken code.